tests/error_translator_test.go

```diff
@@ -27,3 +27,34 @@ func TestDialectorWithErrorTranslatorSupport(t *testing.T) {
 		t.Fatalf("expected err: %v got err: %v", translatedErr, err)
 	}
 }
+
+func TestSupportedDialectorWithErrDuplicatedKey(t *testing.T) {
+	type City struct {
+		gorm.Model
+		Name string `gorm:"unique"`
+	}
+
+	db, err := OpenTestConnection(&gorm.Config{TranslateError: true})
+	if err != nil {
+		t.Fatalf("failed to connect database, got error %v", err)
+	}
+
+	supportedDialectors := map[string]bool{"sqlite": true, "postgres": true, "mysql": true}
```

Comment:
> @a631807682 here is the fix PR for sqlserver driver [go-gorm/sqlserver#103](https://github.com/go-gorm/sqlserver/pull/103), shall we move with this PR?

Is this no longer needed when https://github.com/go-gorm/sqlserver/pull/103 is merged? If so, I expect to wait for https://github.com/go-gorm/sqlserver/pull/103 to be merged